src/librustc/middle/liveness.rs

```diff
@@ -1552,11 +1553,15 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
                     let mut err = self.ir.tcx
                         .struct_span_lint_node(lint::builtin::UNUSED_VARIABLES, id, sp, &msg);
                     if self.ir.variable_is_shorthand(var) {
-                        err.span_suggestion(sp, "try ignoring the field",
-                                            format!("{}: _", name));
+                        err.span_suggestion_with_applicability(sp, "try ignoring the field",
+                                                               format!("{}: _", name),
```

Comment:
I'm not happy with the situation at the moment. Almost every single suggestion needs to have macro checks. I'll be trying something out to encode this in the API and avoid the current game of whack-a-mole.

Reply:
Yeah.

So in clippy my plan was that `span_to_snippet` would take an `&mut applicability` parameter, and whenever it detects a macro in the thing you're using it would automatically flip the applicability.

But also the span you're suggesting for would need a good macro check. idk.